File: Course 04 [Algorithms & Problem-Solving Level 1 Solutions]/My_Solutions_Using_C++/Problem7.cpp
#include <iostream>
using namespace std;

float ReadNumber() {
	float Num;
	cout << "Please enter your number : ";
	cin >> Num;
	return Num;

}

float CalculateHalf(float Num) {
	float HalfNum = Num / 2;
	return HalfNum;
}
void PrintResult(float HalfNum) {
	cout << "the orginal number is " << HalfNum * 2 << " and the Half is " << HalfNum << endl;
}

int main() {
	PrintResult(CalculateHalf(ReadNumber()));
}

File: Course 04 [Algorithms & Problem-Solving Level 1 Solutions]/My_Solutions_Using_C#/Problem7.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Net.Http;
using System.Threading;
using System.Threading.Tasks;

namespace AbuHadHoud
{
    internal class Program
    {
        static int ReadNumber()
        {
            Console.Write("Please enter the number : ");
            int num = Convert.ToInt32(Console.ReadLine());
            return num;
        }
        static double CalculateHalfNumber(int Num)
        {
            return Convert.ToDouble( Num / 2.0);
        }
        static void PrintResult(double HalfNumber)
        {
            Console.WriteLine($"Half of {HalfNumber*2} is {HalfNumber}");
        }

        static void Main(string[] args)
        {
            PrintResult(CalculateHalfNumber(ReadNumber()));
        }
    }
}